scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,980 @@
+    end
+end
+
+local straightEight = function(player, npc, trade)
+    -- First three required
+    if not hasFirstThree(player) then
+        return
+    end
+
+    local initial = string.lower(string.sub(player:getName(), 1, 1)) -- eg. "a"
+    local letter = string.byte(initial) - 97 + xi.items.A_EGG        -- itemID
+    local eggs = {}
+
+    for i = 1, 8 do
+        table.insert(eggs, letter)
+        letter = letter + 1
+
+        if letter > xi.items.Z_EGG then
+            letter = xi.items.A_EGG
+        end
+    end
+
+    if npcUtil.tradeHasExactly(trade, eggs) then
+        if not player:hasItem(xi.items.HAPPY_EGG) then
+            return xi.items.HAPPY_EGG
+        end
+
+        if settings.MINOR_REWARDS then
+            return minorReward()
+        end
+    end
+end
+
+local regionNames =
+{
+    xi.events.egg_hunt.stringToEggs("RONFA"),
+    xi.events.egg_hunt.stringToEggs("ZULKH"),
+    xi.events.egg_hunt.stringToEggs("NORVA"),
+    xi.events.egg_hunt.stringToEggs("GUSTA"),
+    xi.events.egg_hunt.stringToEggs("DERFL"),
+    xi.events.egg_hunt.stringToEggs("SARUT"),
+    xi.events.egg_hunt.stringToEggs("KOLSH"),
+    xi.events.egg_hunt.stringToEggs("ARAGO"),
+    xi.events.egg_hunt.stringToEggs("FAURE"),
+    xi.events.egg_hunt.stringToEggs("VALDE"),
+    xi.events.egg_hunt.stringToEggs("QUFIM"),
+    xi.events.egg_hunt.stringToEggs("LITEL"),
+    xi.events.egg_hunt.stringToEggs("KUZOT"),
+    xi.events.egg_hunt.stringToEggs("VOLLB"),
+    xi.events.egg_hunt.stringToEggs("ELSHI"),
+    xi.events.egg_hunt.stringToEggs("ELSHI"),
+    xi.events.egg_hunt.stringToEggs("TULIA"),
+    xi.events.egg_hunt.stringToEggs("MOVAL"),
+    xi.events.egg_hunt.stringToEggs("TAVNA"),
+}
+
+local beastCostumes =
+{
+    580, -- Yagudo
+    612, -- Orc
+    644, -- Quadav
+}
+
+local nationRewards =
+{
+    { xi.items.WING_EGG,   xi.items.MELODIUS_EGG,  xi.items.EGG_STOOL, },
+    { xi.items.LAMP_EGG,   xi.items.CLOCKWORK_EGG, xi.items.EGG_TABLE, },
+    { xi.items.FLOWER_EGG, xi.items.HATCHLING_EGG, xi.items.EGG_LOCKER, },
+}
+
+local regionControl = function(player, npc, trade)
+    -- First three required
+    if not hasFirstThree(player) then
+        return
+    end
+
+    for regionID, regionName in pairs(regionNames) do
+        if npcUtil.tradeHasExactly(trade, regionName) then
+            local owner = GetRegionOwner(regionID - 1)
+
+            -- Beastmen controlled
+            if owner == 3 then
+                local costume = beastCostumes[math.random(#beastCostumes)]
+                player:addStatusEffect(xi.effect.COSTUME, costume, 0, utils.minutes(60))
+                player:confirmTrade()
+
+                return
+            else
+                local reward = nationRewards[owner + 1]
+
+                -- 2005 Reward
+                if not player:hasItem(reward[1]) then
+                    return reward[1]
+
+                -- 2007 Reward
+                elseif
+                    settings.ERA_2007 and
```

Comment:
Does this mean multiple eras don't stack? Should that be mentioned somewhere?

Reply:
You can turn them on/off individually, they all stack